app/controllers/jobs_controller.rb

```diff
@@ -1,16 +1,29 @@
 class JobsController < ApplicationController
+  before_action :set_company, only: [:new, :create, :edit]
+  before_action :set_job, only: [:show, :edit, :update, :destroy]
```

Comment:
Here especially: for any given controller action I need to now go back here to double check to see what's getting set. In my experience would be better to just have it in the controller action there for me when I needed it.